R/theme_cmap.R

```diff
@@ -178,6 +186,26 @@ theme_cmap <- function(
     ))
   }
 
+  # Introduce x axis ticks if specified
+  if (grepl("x", axisticks)) {
+    add_to_obj(ggplot2::theme(
+      axis.ticks.x = ggplot2::element_line(
```

Comment:
certainly not your fault presuming I'm bet you copied the formatting from the `axislines` code above, but the intention in this function was to use `add_to_obj()` for non-theme elements, and use the named list function for theme elements (See, e.g. `attr[["axis.title.x"]]` in the `xlab` section). 

While the end result is not different, it keeps the results of `theme_cmap()` clean and easy to understand. The result of this function should be a list where the first object is `theme_cmap_base()`, the next n objects are non-theme objects, and then there's a single theme object at the end with any additional theme customization. (see, e.g., `View(theme_cmap(gridlines = "none"))`)

Come to think of it, this line at the end could be only added if `len(attr)>0` to simplify further:
```
# construct final list to return
append(obj, list(do.call(theme, attr)))
```

Let me know if you want to make these changes, or I can jump in and make them.

Reply:
Ah. Yes, I did copy and paste. I can take a look at updating this Monday.

Reply:
It turns out that doing this change actually does impact functionality. If we add things as attributes, if the user then adds an argument that overrides one of those elements (say, `panel.grid.major.y` as I did in the vignette) then it gives an error:
```
Error in (function (line, rect, text, title, aspect.ratio, axis.title,  : 
  formal argument "panel.grid.major.y" matched by multiple actual arguments 
```
Still investigating but flagging in case you know of an obvious solution.

Reply:
Good catch. We'd just need to separate the overrides into their own `theme()` call that comes last.

I guess that means the returned value of `theme_cmap()` should look something like this:

1. n number of non-theme gg objects (hlines, etc)
2. `theme_cmap_base()`
3. `theme(attr)`
4. `theme(...)` where `...` are additional overriding arguments passed to `theme_cmap()`

Right now # 2 happens first, which is not super relevant. The important thing would be the order of the three theme elements.

Reply:
OK I've incorporated this. Please take a look and let me know what you think.